### DISCUSSION TITLE
Looking for a OSDP PACS reader with smart card support

### DISCUSSION THREAD

**Discussion by CardContact** [goToMain/libosdp#207] [Q&A]
We are the manufacturer of the the SmartCard-HSM, a PKI token which is available as contactless card with ISO 14443-4 interface. The card supports a standardized authentication protocol based on Chip Authentication V2 and Card Verifiable Certificates. Basically the same protocol that is used in Electronic Passports (google for BSI TR03110 for details).

As we now (again) would like to promote the [coolPACS](https://www.coolpacs.org) idea from 2014, we are looking for a PACS reader with PIN Pad that supports a transparent access to a smart card using ISO 7816-4 APDUs.

Our idea is to use a Raspberry based controller to implement the PKI based authentication protocol. The reader therefore needs to provide transparent access to the card.

Any idea which reader with OSDP support would fit ?

**Comment by sidcha** [comment]
IIRC, HID has a reader with keypad and can do a form of transparent access (which is not blessed by SIA?). When I was at Honeywell, we added support for 14443 Type-B cards so some of their readers (can't recall the model but I can find out) in theory could work.

Bigger problem would be, LibOSDP does not have support for transparent mode. If you are planning on doing any work in this regard, please feel free to reach out to me to explore possible ways to integrate the support here (as you can see in #22, I am very keen on landing this feature).

**Comment by sidcha** [comment]
The way I see it, the best course of action would be to use some dev-kit PD setup (of which there are many) to add support for transparent reader support to LibOSDP and then use that with the Raspberry Pi controller. This gives you maximum debug-ability when things go south.

**Comment by sidcha** [comment]
Closing discussion as stale.
